src/api/cvc4cpp.cpp

```diff
@@ -34,6 +35,46 @@
 namespace CVC4 {
 namespace api {
 
+namespace {
+/* -------------------------------------------------------------------------- */
+/* API guard helpers                                                          */
+/* -------------------------------------------------------------------------- */
+
+class CVC4_PUBLIC CVC4ApiException : public Exception
```

Comment:
We probably need to define this exception in the header s.t. people can catch it. I don't think that it necessarily needs to inherit from `CVC4::Exception` but can directly inherit from `std::exception`.

Reply:
That's what I'm currently doing, this PR will change quite a bit, just FYI.